### ISSUE TITLE
Improve Language switcher look

### ISSUE THREAD

**Issue by CyrilWendl** [warmshowers/Warmshowers.org#888]
I would like the Language switcher look to be improved, in order to be more visible and prominent, especially since the flags aren't visible anymore in D7. I would also like to slightly increase the size of the language switcher. Language symbols are probably inadequate, as most languages are spoken in several countries. I like a lot the [Avaaz](https://avaaz.org/en/) approach, showing every language in the header (might be overkill though). I like the Facebook approach too, with the languages in the footer.

**Comment by rfay** [comment]
Interested in your views on this, @jeanfrancoisbeaulieu

**Comment by jeanfrancoisbeaulieu** [comment]
I think the Avaaz approach is overkill, especially because most user will change only once on WS. But I agree that it's not very pretty right now. I also agree that flags aren't adequate.

Facebook approach is interesting, displaying most common languages and a 'more' button. Also it will only display when you aren't logged in, which is a good idea imo. Footer position is also nice, but that means your page needs to be short.

You can assign this to me.

**Comment by rfay** [comment]
Assigned to @jeanfrancoisbeaulieu - but remember there may be flag needs like the setup we had before. Or a different widget.

**Comment by jeanfrancoisbeaulieu** [comment]
Could be just a nicely styled dropdown, or something like this. Just options,not sure yet.
![image](https://cloud.githubusercontent.com/assets/9020402/10559921/a8c5afb8-74c8-11e5-93db-b74428fefa1e.png)
This kind of visual would only appear on the anon dashboard thought.

I would be in favor of having the switcher/language in the header, only when not logged in. Maybe then it could be in the footer? What your opinion on this @rfay ?

**Comment by CyrilWendl** [comment]
This looks *clean*! :-) Is this already part of the redesign project?

**Comment by rfay** [comment]
Remember that the language switch is something that should be do-able on any page.

I'd like it to be really obvious too, because remember what it looks like to stare at a page that is not in your own language. You need to find the way in quickly.

But I like the look as well!

**Comment by jeanfrancoisbeaulieu** [comment]
Once you are logged in, is it really necessary? I assume that once you are logged, 99% of the pages you will look at will be in your own language. Maybe you will follow a link to a different version or something.. (from the forum? or facebook?)  but then we could detect if it's different from your settings language and show a message to switch. 

Imo, most user will not use the switcher after login. But maybe there is some use cases I'm not thinking about? Just thoughts here.

**Comment by rfay** [comment]
Absolutely, yes. And people end up on the wrong language lots of the time. 

The current switcher is specifically designed to show when there is in fact a replica of the current page available in a different language.

**Comment by jeanfrancoisbeaulieu** [comment]
How do they manage to do that? :) 

For me, the best way to keep it clean and make it a good experience would be to: 
- Have a clear and obvious switcher/list on the anon homepage.
- Have a switcher in the footer once you're logged in
- If logged in and accessing a page in a language different then the one in your own language, have a message pop up saying something like "this page is also available in 'your language' would you like to change/see it".

In the end it's mostly because I find the switcher takes space in the header and *I* don't use it. But, it's not that bad I guess.

**Comment by rfay** [comment]
You forget the most important pages of all, user/, /user/password and
user/register - pages which anon people access in English for various
reasons but it's *critical* to switch languages.

In addition, people often land on the wrong language through things like
email notifications, etc. Currently (and forever) forum activity
notifications have gone out in the language (and with the URLs) of the
*commenting* user, not the appropriate recipient. I'd love to fix this but
it's down the line.

This is not just a design issue, it's an overall user experience issue.

I would definitely be interested in capturing actual user sequences, as
that would help us a lot. But I'm not sure how to capture that kind of info.

But at this point I definitely wouldn't approve any solution that doesn't
allow seamless switching anywhere on the site.

**Comment by rfay** [comment]
Oh, and please note that switching means switching to the correct corresponding URL in the related language. Most of the time that's the same URL, but not always.

**Comment by jeanfrancoisbeaulieu** [comment]
I sure agree for user/ and those pages, very important. Actually anytime you aren't logged in.

For the rest, it's just that I wouldn't put it as prominent when logged in, and have a message reminding you to switch. I just think that would be a nice user experience, but it doesn't have to be that really. 

Would be nice for user sequences, would be very interesting. Google Analitycs can probably help but we'd have to put more hook I think. Haven't done that before.

Anyway, I can just re-skin the switcher, I thought it was interesting to reflect on it.

**Comment by rfay** [comment]
It's worth considering fixing the languageicons module too so we can have
the flags. Sometimes they solve some things visually.

**Comment by jeanfrancoisbeaulieu** [comment]
I don't like the flags much cause I don't like to associate a language with a country, but they sure provide a visual clue.

**Comment by rfay** [comment]
Yes, and there is always somebody from Brazil complaining about the
Portuguese flag!

**Comment by CyrilWendl** [comment]
I don't like flags either, as I have pointed out in my initial comment. I'm rather following @jeanfrancoisbeaulieu 's thoughts on keeping the language switcher as simple as possible once a user is logged in. I also think we have to make major efforts that users stay on their language, and think it would be worth opening another issue for this. I would like:
- to redirect users always to their language by using the language of their browser.
- to use any other techniques to recognize their language
- to make major efforts in all translations to assure that all links redirect to the same language. This involves checking all important pieces of content (/translation/status) in all languages. I might notify translators to do this in an upcoming mail, which will also contain D7 instructions.

**Comment by CyrilWendl** [comment]
For this issue, I think we could just show the languages on every page and eventually simplify them later. Also, I really like the look of the languages links on @jeanfrancoisbeaulieu 's screenshot.

**Comment by kalpaitch** [comment]
Quite happy considering @jeanfrancoisbeaulieu's design, whatever we do it will probably have to be built ourselves, the previous language switcher icons thing don't be workin no more.

**Comment by kalpaitch** [comment]
@jeanfrancoisbeaulieu I'm happy with how the language switcher looks now, can we close this ticket?

**Comment by CyrilWendl** [comment]
Looks great indeed! I think this issue can be closed. Don't hesitate to reopen it if you like.
